features/dd-sdk-android-trace/src/test/kotlin/com/datadog/android/trace/internal/data/OtelTraceWriterTest.kt

```diff
@@ -301,5 +307,21 @@ internal class OtelTraceWriterTest {
         verifyNoMoreInteractions(mockSdkCore, mockTracingFeatureScope)
     }
 
+    private fun createNonEmptyDdSpans(forge: Forge, isDropSampling: Boolean): MutableList<DDSpan> {
+        val predicate: (DDSpan) -> Boolean = if (isDropSampling) {
+            { it.samplingPriority() in OtelTraceWriter.DROP_SAMPLING_PRIORITIES }
+        } else {
+            { it.samplingPriority() !in OtelTraceWriter.DROP_SAMPLING_PRIORITIES }
+        }
+
+        var spans = emptyList<DDSpan>()
+        while (spans.isEmpty()) {
+            spans = forge.aList { getForgery<DDSpan>() }
+                .filter(predicate)
+        }
```

Comment:
we can also use approach which @aleksandr-gringauz did in https://github.com/DataDog/dd-sdk-android/pull/2565 (there it is a bit more flexible), but here it is also okay